modules/core/src/main/java/org/apache/synapse/endpoints/auth/oauth/AuthorizationCodeHandler.java

```diff
@@ -69,6 +71,18 @@ protected OMElement serializeSpecificOAuthConfigs(OMFactory omFactory) {
 
         authCode.addChild(
                 OAuthUtils.createOMElementWithValue(omFactory, AuthConstants.OAUTH_REFRESH_TOKEN, getRefreshToken()));
+        if (connectionTimeout != -1) {
+            authCode.addChild(OAuthUtils.createOMElementWithValue(omFactory, AuthConstants.OAUTH_CONNECTION_TIMEOUT,
+                    String.valueOf(connectionTimeout)));
+        }
+        if (connectionRequestTimeout != -1) {
+            authCode.addChild(OAuthUtils.createOMElementWithValue(omFactory, AuthConstants.OAUTH_CONNECTION_REQUEST_TIMEOUT,
+                    String.valueOf(connectionRequestTimeout)));
+        }
+        if (socketTimeout != -1) {
+            authCode.addChild(OAuthUtils.createOMElementWithValue(omFactory, AuthConstants.OAUTH_SOCKET_TIMEOUT,
+                    String.valueOf(socketTimeout)));
+        }
```

Comment:
Since these configs are common for all the OAuth types, let's move this logic to the `serializeOAuthConfiguration` [1] method in the `OAuthHandler` parent class.

[1] - https://github.com/wso2/wso2-synapse/blob/master/modules/core/src/main/java/org/apache/synapse/endpoints/auth/oauth/OAuthHandler.java#L181

Reply:
Fixed